node/tests/SharedTests.ts

```diff
@@ -2367,6 +2367,38 @@ export function runBaseTests<Context>(config: {
         config.timeout,
     );
 
+    it.only.each([ProtocolVersion.RESP2, ProtocolVersion.RESP3])(
+        "renamenx test_%p",
+        async (protocol) => {
+            await runTest(async (client: BaseClient) => {
+                const key1 = `{key}-1-${uuidv4()}`;
+                const key2 = `{key}-2-${uuidv4()}`;
+                const key3 = `{key}-3-${uuidv4()}`;
+
+                // renamenx missing key
+                try {
+                    expect(await client.renamenx(key1, key2)).toThrow();
+                } catch (e) {
+                    expect((e as Error).message).toMatch("no such key");
+                }
+
+                // renamenx a string
+                await client.set(key1, "key1");
+                await client.set(key3, "key3");
+                expect(await client.renamenx(key1, key2)).toEqual(true);
+                expect(await client.renamenx(key2, key3)).toEqual(false);
+                const result2 = await client.get(key2);
+                expect(result2).toEqual("key1");
+                await client.del([key1, key2]);
```

Comment:
I don't understand why we are deleting these?

Reply:
I follow Yury's Java implementation (SharedCommandTests.java).
I agree that we don't need this.